### PR TITLE
CUMULUS-3160: Upgrade to knex 2.4.1 and jsonwebtoken 9.0.0 on release11.1.x

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/spec/parallel/cnmWorkflow/LambdaEventSourceSpec.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ describe('When adding multiple rules that share a kinesis event stream', () => {
     // clean up stack state added by test
     console.log(`\nCleaning up stack & deleting test stream '${streamName}'`);
     await deleteRules(testConfig.stackName, testConfig.bucket, rulesToDelete, ruleSuffix);
-    await deleteExecution({ prefix: testConfig.stackName, executionArn: executionArn });
+    await waitForExecutionAndDelete(testConfig.stackName, executionArn, 'completed');
```
Comment (npauzenga) [inline example/spec/parallel/cnmWorkflow/LambdaEventSourceSpec.js:73]: Are these the same test fixes that you made in main? Either way, thanks for handling them.

File: package.json [hunk 2]
```diff
@@ -152,7 +152,7 @@
     "supertest": "^4.0.2",
     "tough-cookie": "^4.0.0",
     "tsc-watch": "^4.2.8",
-    "typescript": "~4.7.3",
+    "typescript": "~4.8.3",
```
Comment (npauzenga) [inline package.json:155]: We need the TS update for knex v2.4.1? I assume so, just checking.

File: packages/integration-tests/api/distribution.js [hunk 3]
```diff
@@ -231,6 +231,7 @@ async function getTEARequestJwtToken(
     exp: expirationTime,
   }, jwtPrivateKey, {
     algorithm: JwtAlgo,
+    allowInsecureKeySizes: true,
```
Comment (npauzenga) [inline packages/integration-tests/api/distribution.js:234]: What's this?

File: audit-ci.json [hunk 4]
```diff
@@ -2,5 +2,5 @@
     "high": true,
     "pass-enoaudit": true,
     "retry-count": 20,
-    "allowlist": ["node-forge", "xmldom"]
+    "allowlist": ["knex", "jsonwebtoken"]
```
[the remark this answers is not in the document]
Comment (npauzenga) [reply audit-ci.json:5]: Nevermind. I see your reply to a similar comment on https://github.com/nasa/cumulus/pull/3222

File: example/spec/parallel/cnmWorkflow/LambdaEventSourceSpec.js (hunk 1 above)
Comment (npauzenga) [reply example/spec/parallel/cnmWorkflow/LambdaEventSourceSpec.js:73]: Yeah, seems like you needed to make all of these changes with the knex 2.x update. Same as https://github.com/nasa/cumulus/pull/3222

**Review by npauzenga — APPROVED** [review]
Looks great @jennyhliu! Thanks for handling this.

File: package.json (hunk 2 above)
Comment (jennyhliu) [reply package.json:155]: Yes, make it consistent with what knex v2.4.1 is using

File: packages/integration-tests/api/distribution.js (hunk 3 above)
Comment (jennyhliu) [reply packages/integration-tests/api/distribution.js:234]: It's part of the jsonwebtoken v8 to v9 migration. The length of the key we are using is less than the new required length.
https://github.com/nasa/cumulus/pull/3203/files#r1055865223

**Pull request merged**